client/data/agency-dashboard/use-sync-jetpack-connection-status.ts

```diff
@@ -0,0 +1,61 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useContext, useEffect, useMemo } from 'react';
+import SitesOverviewContext from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/context';
+import { Site } from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/types';
+import wpcom from 'calypso/lib/wp';
+
+const useSyncJetpackConnectionStatus = (
+	isPartnerOAuthTokenLoaded: boolean,
+	isConnectionHealthy: boolean,
+	siteId: number
+) => {
+	const queryClient = useQueryClient();
+	const { search, currentPage, filter, sort } = useContext( SitesOverviewContext );
+
+	const dashboardSitesQueryKey = useMemo(
+		() => [ 'jetpack-agency-dashboard-sites', search, currentPage, filter, sort ],
+		[ currentPage, filter, search, sort ]
+	);
+
+	const { data } = useQuery( {
+		queryKey: [ 'jetpack-agency-test-connection', { siteId, isConnectionHealthy } ],
```

Comment:
This looks like it's a generic jetpack connection test query, so I'd be inclined to remove the `agency` from the key.